inference-engine/thirdparty/clDNN/src/batch_to_space.cpp

```diff
@@ -34,88 +34,56 @@ layout batch_to_space_inst::calc_output_layout(batch_to_space_node const& node)
-        std::vector<int32_t> sizes;
-        if (input.get_output_layout().data_type == cldnn::data_types::i64) {
-            int64_t* data = static_cast<int64_t*>(mem.lock());
-            std::vector<int64_t> sizes_i64 = std::vector<int64_t>(data, data + input.get_output_layout().count());
-            sizes.resize(sizes_i64.size());
-            for (size_t j = 0; j < sizes.size(); j++)
-                sizes[j] = static_cast<int32_t>(sizes_i64[j]);
-        } else {
-            int32_t* data = static_cast<int32_t*>(mem.lock());
-            sizes = std::vector<int32_t>(data, data + input.get_output_layout().count());
-        }
-        args.push_back(sizes);
-        mem.unlock();
+    if (output_format == format::b_fs_yx_fsv16) {
+        output_format = format::bfyx;
     }
+    auto output_type = desc->output_data_type ? *desc->output_data_type : input_layout.data_type;
+
+    if (node.has_fused_primitives())
+        output_type = node.get_fused_output_layout().data_type;
+
+    const size_t dims_num = format::dimension(input_format);
 
-    const auto& block_shape = args[0];
-    const auto& crops_begin = args[1];
-    const auto& crops_end = args[2];
+    const auto& block_shape = desc->block_shape;
+    const auto& crops_begin = desc->crops_begin;
+    const auto& crops_end = desc->crops_end;
 
-    if (block_shape[0] != 1)
+    if (block_shape.batch[0] != 1)
         CLDNN_ERROR_MESSAGE(node.id(),
             "block_shape[0] is expected to be 1. Actual block_shape[0] is " +
-            std::to_string(block_shape[0]));
+            std::to_string(block_shape.batch[0]));
 
-    if (crops_begin[0] != 0)
+    if (crops_begin.batch[0] != 0)
         CLDNN_ERROR_MESSAGE(node.id(),
             "crops_begin[0] is expected to be 0. Actual crops_begin[0] is " +
-            std::to_string(crops_begin[0]));
+            std::to_string(crops_begin.batch[0]));
 
-    if (crops_end[0] != 0)
+    if (crops_end.batch[0] != 0)
         CLDNN_ERROR_MESSAGE(node.id(),
             "crops_end[0] is expected to be 0. Actual crops_end[0] is " +
-            std::to_string(crops_end[0]));
+            std::to_string(crops_end.batch[0]));
 
-    size_t block_sizes_multiplied = 1;
-    for (const auto& el: block_shape)
-        block_sizes_multiplied *= el;
+    size_t block_sizes_multiplied = block_shape.feature[0];
+    for (size_t i = 0; i < dims_num - 2; ++i)
+        block_sizes_multiplied *= block_shape.spatial[i];
 
     if (input_layout.size.batch[0] % block_sizes_multiplied != 0)
         CLDNN_ERROR_MESSAGE(node.id(),
             "The batch of the input tensor must be divisible by multiplied block sizes = " +
             std::to_string(block_sizes_multiplied));
 
-    for(size_t i = 1; i < dims_num; ++i)
-        if (crops_begin[i] + crops_end[i] >= block_shape[i] * input_layout.size.sizes(input_format)[i])
+    if (crops_begin.feature[0] + crops_end.feature[0] >= block_shape.feature[0] * input_layout.size.sizes(input_format)[1])
```

Comment:
As you use `input_layout.size.sizes(format)` function, returned values will ordered regarding to format, so in `bfyx` layout Feature will be at [1] index, but not in `yxfb`.